L7-JUnit Params, Mockito/L7-Zad.7-Training/src/main/java/pl/babiarz/TrainingEffectiveness.java

```diff
@@ -0,0 +1,20 @@
+package pl.babiarz;
+
+public class TrainingEffectiveness {
+
+    TrainingLength trainingLength = new TrainingLength();
+    CaloriesBurned caloriesBurned = new CaloriesBurned();
+    AveragePulseRate averagePulseRate = new AveragePulseRate();
```

Comment:
Skoro już wiemy jak działa Mockito to może widzisz tutaj zastosowanie? 

Jeżeli stworzysz te obiekty przekazujac je przez konstruktor, a następnie w tescie (TraningEffectivenessTest) je zamockujesz to będziesz w stanie przetestować tę klasę Training[...] bez zalezności do tych 3 klas.

Podobnie jak to zrobiłeś tutaj:
https://github.com/SzymonB7/JavaIntro_Zadania/blob/master/L7-JUnit%20Params%2C%20Mockito/L7-Zad.6-CurrencyCalculator/src/test/java/CurrencyCalculatorTest.java

Możesz założyć sobie, że np. trainingLength.getTrainingLengthEvaluation(min); zwraca jakąś wartość

caloriesBurned.getCaloriesBurnedEvaluation(kcal); jakąś wartość itd i na tej podstawie sprawdzasz czy na pewno TrainingEffectiveness  zachowuje sie poprawnie

Reply:
ale przydadzą się interfejsy ;)